native/cocos/core/scene-graph/Node.cpp

```diff
@@ -606,27 +574,17 @@ Mat4 Node::getWorldRT() {
     return target;
 }
 
-void Node::invalidateChildren(TransformBit dirtyBit) {
+void Node::invalidateChildren(TransformBit dirtyBit) { //NOLINT(misc-no-recursion)
     auto curDirtyBit{static_cast<uint32_t>(dirtyBit)};
-    const uint32_t childDirtyBit{curDirtyBit | static_cast<uint32_t>(TransformBit::POSITION)};
-    setDirtyNode(0, this);
-    int i{0};
-    while (i >= 0) {
-        Node *cur = getDirtyNode(i--);
-        if (cur == nullptr) {
-            continue;
+    const uint32_t hasChangedFlags = getChangedFlags();
+    const uint32_t dirtyFlags = getDirtyFlag();
+    if (isValid() && (dirtyFlags & hasChangedFlags & curDirtyBit) != curDirtyBit) {
+        setDirtyFlag(dirtyFlags | curDirtyBit);
+        setChangedFlags(hasChangedFlags | curDirtyBit);
+
+        for (Node *child : getChildren()) {
+            child->invalidateChildren(dirtyBit | TransformBit::POSITION);
```

Comment:
why add TransformBit::POSITION

Reply:
The original procedure did this too. 

If a node update it's scale or rotation, all children's position are also effected.